fastly/s3.go

```diff
@@ -217,6 +231,7 @@ type UpdateS3Input struct {
 	PublicKey                    *string                `form:"public_key,omitempty"`
 	ServerSideEncryptionKMSKeyID *string                `form:"server_side_encryption_kms_key_id,omitempty"`
 	ServerSideEncryption         S3ServerSideEncryption `form:"server_side_encryption,omitempty"`
+	ACL                          S3AccessControlList    `form:"acl,omitempty"`
```

Comment:
I think this might need to be a pointer to the type? Although, that said `S3ServerSideEncryption` isn't a pointer 🤔

Reply:
I thought that as well, as `S3Redundancy` also isn't a pointer 🤔 I believe the pointer is redundant for all of them actually due to `omitempty` but I just tried to fit in with the existing style - should I change it?

Reply:
Well iirc the reason for using pointers was to allow for checking if something was a zero value or genuinely not provided (because the zero value for a pointer would be nil) and so I feel like it should be changed. In the go-fastly README we even link to a really good article about it: https://willnorris.com/2014/05/go-rest-apis-and-pointers/

Reply:
Yes you're right - `omitempty` and _not_ having a pointer means you can't `UPDATE` the value to be `""`, I'll change it